module/VuFindSearch/src/VuFindSearch/Backend/Primo/Response/RecordCollectionFactory.php

```diff
@@ -50,6 +57,32 @@ protected function getDefaultRecordCollectionClass(): string
         return RecordCollection::class;
     }
 
+    /**
+     * Return record collection.
+     *
+     * @param array $response Deserialized JSON response
+     *
+     * @return RecordCollection
+     */
+    public function factory($response)
+    {
+        if (!is_array($response)) {
+            throw new InvalidArgumentException(
+                sprintf(
+                    'Unexpected type of value: Expected array, got %s',
+                    gettype($response)
+                )
+            );
+        }
+        $collection = new $this->collectionClass($response);
+        foreach ($response['documents'] ?? [] as $doc) {
+            $record = call_user_func($this->recordFactory, $doc);
```

Comment:
Could this just be:

```suggestion
            $record = ($this->recordFactory)($doc);
```

...to eliminate the need for `call_user_func`?